### PR TITLE
Docs/capigw 040 route details prefix rewrite

### PR DESCRIPTION
**Pull request by trujillo-adam**
### Description
This PR completes the Route configuration reference and adds some usage information for rewriting prefix paths for incoming HTTP requests.

### Links
[Original draft](https://docs.google.com/document/d/1EyjIUFwu7WcUate0YNWmO4_dFR5KiAWHLk6spVAyi7U/)

### PR Checklist

* [ ] updated test coverage
* [X] external facing docs updated
* [ ] not a security concern

### CODE REVIEW

**Comment by trujillo-adam** [comment]
There are still several gaps on the Routes configuration reference page that I need help filling in.

File: website/content/docs/api-gateway/configuration/routes.mdx [hunk 1]
```diff
@@ -78,6 +166,105 @@ The following example creates a route named `example-route` in namespace `gatewa
...
+* [query parameters](#rules-matches-queryparams)
+* [request method](#rules-matches-method)
+
+### rules.matches.path
+
+Specifies a list of objects that define matches based on URL path. The following table describes the parameters for the `path` field:
+
+| Parameter | Description | Type | Required |
+| ---       | ---         | ---  | ---      |
+| `type` | Specifies the type of comparison to use for matching the path value. You can specify the following types. <ul><li>`Exact`: Returns a match only when the entire path matches the `value` field (default).</li><li> `PathPrefix`: Returns a match when the path matches the regex defined in the `value` field.</li></ul> | String | Required |
```
Comment (trujillo-adam) [inline website/content/docs/api-gateway/configuration/routes.mdx:230]:
```suggestion
| `type` | Specifies the type of comparison to use for matching the path value. You can specify the following types. <ul><li>`Exact`: Returns a match only when the entire path matches the `value` field (default).</li><li> `PathPrefix`: Returns a match when the path matches the regex defined in the `value` field.</li></ul> | String | Required |
```
@nathancoleman - Should the `PathPrefix` type actually be `RegularExpression` like the other matching fields? I wasn't sure if this was just an error.

File: website/content/docs/api-gateway/configuration/routes.mdx [hunk 2]
```diff
@@ -78,6 +166,105 @@ The following example creates a route named `example-route` in namespace `gatewa
...
+
+The following table describes the parameters for `path`:
+
+| Parameter | Description | Type | Required |
+| ---       | ---         | ---  | ---      |
+| `replacePrefixMatch` | Specifies a value that replaces the path prefix for incoming HTTP requests. The operation only affects the path prefix. The rest of the path is unchanged. | String | Required |
+| `type` | Specifies the type of replacement to use for the URL path. You can specify the following values: <ul><li>`ReplacePrefixMatch`: Replaces the the matched prefix of the URL path (default). </li></ul> | String | Optional |
+
+### rules.matches
+
```
Comment (trujillo-adam) [inline website/content/docs/api-gateway/configuration/routes.mdx:216]: I assume that you would be able to define several of each type of matching rule if you needed to narrow the matches down. If so, should we explicitly say so? Or is it reasonable for our audience to make that assumption?

File: website/content/docs/api-gateway/configuration/routes.mdx [hunk 3]
```diff
@@ -78,6 +166,105 @@ The following example creates a route named `example-route` in namespace `gatewa
...
+* [request method](#rules-matches-method)
+
+### rules.matches.path
+
+Specifies a list of objects that define matches based on URL path. The following table describes the parameters for the `path` field:
+
+| Parameter | Description | Type | Required |
+| ---       | ---         | ---  | ---      |
+| `type` | Specifies the type of comparison to use for matching the path value. You can specify the following types. <ul><li>`Exact`: Returns a match only when the entire path matches the `value` field (default).</li><li> `PathPrefix`: Returns a match when the path matches the regex defined in the `value` field.</li></ul> | String | Required |
+| `value` | Specifies value to match on. You can specify a specific string or a regular expression. | String | Required |
```
Comment (nathancoleman) [inline website/content/docs/api-gateway/configuration/routes.mdx:231]:
```suggestion
| `value` | Specifies value to match on. You can specify a specific string for `Exact` and `PathPrefix` types or a regular expression for the `RegularExpression` type. | String | Required |
```

**Review by nathancoleman — CHANGES_REQUESTED** [review]
Responding to your comments, thanks for porting this over!

Is there reason to keep the GDocs version in sync with what winds up merging here, or are you intending to throw that doc away on merge?

**Comment by trujillo-adam** [comment]
> Responding to your comments, thanks for porting this over!
> 
> Is there reason to keep the GDocs version in sync with what winds up merging here, or are you intending to throw that doc away on merge?

No reason to keep the Gdoc. The purpose of the GDoc is/was to draft and collaborate more efficiently and minimize all of the back and forth in GH. We can ignore it now that 95% of the content is finalized.

**Review by trujillo-adam — COMMENTED** [review]
One suggestion for the MeshService page.

File: website/content/docs/api-gateway/configuration/routes.mdx [hunk 4]
```diff
@@ -78,12 +166,100 @@ The following example creates a route named `example-route` in namespace `gatewa
...
+
+### rules.matches
+
+Specifies rules for matching incoming requests. You can apply [`filters`](#rulesfilters) to requests that match the defined rules. You can match incoming requests based on the following elements:
+
+* [paths](#rules-matches-path)
+* [headers](#rules-matches-headers)
+* [query parameters](#rules-matches-queryparams)
+* [request method](#rules-matches-method)
+
```
Comment (trujillo-adam) [inline website/content/docs/api-gateway/configuration/routes.mdx:223]: Each rule matches requests independently. As a result, a request matching any of the conditions is considered a match.  You can configure several matching rules for each type to widen or narrow matches.

Comment (trujillo-adam) [inline website/content/docs/api-gateway/configuration/routes.mdx:223]:
```suggestion
Each rule matches requests independently. As a result, a request matching any of the conditions is considered a match.  You can configure several matching rules for each type to widen or narrow matches.
```

File: website/content/docs/api-gateway/configuration/routes.mdx [hunk 5]
```diff
@@ -78,12 +166,100 @@ The following example creates a route named `example-route` in namespace `gatewa
...
+### rules.matches
+
+Specifies rules for matching incoming requests. You can apply [`filters`](#rulesfilters) to requests that match the defined rules. You can match incoming requests based on the following elements:
+
+* [paths](#rules-matches-path)
+* [headers](#rules-matches-headers)
+* [query parameters](#rules-matches-queryparams)
+* [request method](#rules-matches-method)
+Each rule matches requests independently. As a result, a request matching any of the conditions is considered a match.  You can configure several matching rules for each type to widen or narrow matches.
```
Comment (nathancoleman) [inline website/content/docs/api-gateway/configuration/routes.mdx:223]:
```suggestion
* [request method](#rules-matches-method)

Each rule matches requests independently. As a result, a request matching any of the conditions is considered a match.  You can configure several matching rules for each type to widen or narrow matches.
```

File: website/content/docs/api-gateway/configuration/routes.mdx [hunk 6]
```diff
@@ -78,12 +166,100 @@ The following example creates a route named `example-route` in namespace `gatewa
...
+### rules.matches.path
+
+Specifies a list of objects that define matches based on URL path. The following table describes the parameters for the `path` field:
+
+| Parameter | Description | Type | Required |
+| ---       | ---         | ---  | ---      |
+| `type` | Specifies the type of comparison to use for matching the path value. You can specify the following types. <ul><li>`Exact`: Returns a match only when the entire path matches the `value` field (default).</li><li> `PathPrefix`: Returns a match when the path matches the regex defined in the `value` field.</li></ul> | String | Required |
```
Comment (nathancoleman) [inline website/content/docs/api-gateway/configuration/routes.mdx:230]:
```suggestion
| `type` | Specifies the type of comparison to use for matching the path value. You can specify the following types. <ul><li>`Exact`: Returns a match only when the entire path matches the `value` field (default).</li><li> `PathPrefix`: Returns a match when the path has the prefix defined in the `value` field.</li><li>`RegularExpression`: Returns a match when the path matches the regex defined in the `value` field.</li></ul> | String | Required |
```

**Review by nathancoleman — CHANGES_REQUESTED** [review]
These are the last remaining things that stick out for me code-wise. Gonna take one last look at the Vercel preview before approving

File: website/content/docs/api-gateway/configuration/meshservice.mdx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+  This topic describes how to configure the Consul API Gateway MeshService object
+---
+
+# MeshService
+
+This topic provides full details about the `MeshService` resource.
+
+## Introduction
+
+A `MeshService` is a resource in the Kubernetes cluster that represents a service in the Consul service mesh outside the Kubernetes cluster where Consul API Gateway is deployed. The service must be in the same Consul datacenter. The MeshService exists so that other configuration models in Kubernetes, such as HTTPRoute and TCPRoute, can reference services that only exist in Consul.
```
[the remark this answers is not in the document]
Comment (nathancoleman) [reply website/content/docs/api-gateway/configuration/meshservice.mdx]: Love this change!

File: website/content/docs/api-gateway/configuration/routes.mdx [hunk 8]
```diff
@@ -78,12 +166,101 @@ The following example creates a route named `example-route` in namespace `gatewa
...
+
+| Parameter | Description | Type | Required |
+| ---       | ---         | ---  | ---      |
+| `set` | Configure this field to rewrite the HTTP request header. It specifies the name of an HTTP header to overwrite and the new value to set. Any existing values associated with the header name are overwritten. You can specify the following configurations: <ul><li>`name`: Required string that specifies the name of the HTTP header to set.</li><li>`value`: Required string that specifies the value of the HTTP header to set.</li></ul> | List of objects | Optional |
+| `add` | Configure this field to append the request header with a new value. It specifies the name of an HTTP header to append and the value(s) to add. You can specify the following configurations: <ul><li>`name`: Required string that specifies the name of the HTTP header to append.</li><li>`value`: Required string that specifies the value of the HTTP header to add.</li></ul> | List of objects | Optional |
+| `remove` | Configure this field to specifify an array of header names to remove from the request header. | Array of strings | Optional |
```
Comment (nathancoleman) [inline website/content/docs/api-gateway/configuration/routes.mdx:194]:
```suggestion
| `remove` | Configure this field to specify an array of header names to remove from the request header. | Array of strings | Optional |
```

**Review by nathancoleman — APPROVED** [review]

**Pull request merged**